moto/iot/models.py

```diff
@@ -104,10 +104,103 @@ def matches(self, query_string: str) -> bool:
                     return True
             return False
         if query_string.startswith("attributes."):
-            k, v = query_string[11:].split(":")
-            return self.attributes.get(k) == v
+            try:
+                k, v = query_string[11:].split(":", 1)
+                return self.attributes.get(k) == v
+            except ValueError:
+                return False
         return query_string in self.thing_name
 
+    def matches(self, query_string: str) -> bool:
+        if not query_string.strip():
+            return False
+        if query_string == "*":
+            return True
+
+        # Replace "-" sign at the beginning of a term with a regular NOT operator
```

Comment:
Appreciate the occasional comment here - that makes a lot easier to understand :slightly_smiling_face: